lib/dataset/bin_detail.cpp

```diff
@@ -28,17 +28,26 @@ Variable make_range(const scipp::index begin, const scipp::index end,
                 dim, CumSumMode::Exclusive);
 }
 
+namespace {
+template <class T> Variable as_edge_view(T &edges) {
+  if (is_bins(edges)) {
+    return as_subspan_view(edges);
+  }
+  const Dim dim = edges.dims().inner();
+  const auto con_edges = scipp::variable::as_contiguous(edges, dim);
+  return subspan_view(con_edges, dim);
+}
+} // namespace
```

Comment:
Regarding your failing test: See the big warning in `histogram.cpp` about `as_contiguous` returning a new variable. Aren't you running into the same problem here?